imgts/entrypoint.sh

```diff
@@ -37,11 +37,69 @@ ERRIMGS=''
 # It's possible for multiple simultaneous label updates to clash
 CLASHMSG='Labels fingerprint either invalid or resource labels have changed'
 
+# In an effort to avoid unintentional deletion of release-branch VM images
+# this image-use context must be detected.  This function accepts a single
+# argument: the Cirrus-CI build ID.  It attempts to determine if that build
+# occured on a non-main branch, and if so will return zero.  Otherwise,
+# it will return non-zero for executions on behalf of all PRs or tags.
+is_release_branch_image(){
+    local buildId api query result fltrpfx branch tag
+    buildId=$1
+    api="https://api.github.com/graphql"
+    query="{
+        \"query\": \"query {
+            build(id: $buildId) {
+                branch
+                tag
+                pullRequest
+              }\"
+        }"
+
+    # It's possible for an image to be missing it's build ID label.
+    # For example, the first time imgts operates on a new image.
+    if ((${#buildId}<12)); then
+        warn 0 "Empty/invalid BuildId value found on image, ignoring: '$buildId'"
+        return 1
+    fi
+
+    fltrpfx=".data.build"
+    result=$(curl --silent --location \
+             --request POST --data @- --url "$api" <<<"$query") \
+             || \
+             die 3 "Error communicating with GraphQL API $api: $result"
+
+    # Best effort: It's possible GraphQL query errored, is invalid,
+    # or the build is so old, there is no record of it.  Issue a
+    # warning and move on.
+    if ! jq -e "$fltrpfx" <<<"$result" &> /dev/null; then
+        warn 0 "Received unexpected reply: $result"
```

Comment:
Anyone reading this log is going to know what JSON is, but will not necessarily know what "expected" or "unexpected" is. How about `Response from github does not validate as JSON '$query_root': $result`

Reply:
It's the Cirrus API, but sure I can change it.